### PR TITLE
Use cluster scripts from kubevirtci repo

### PR DESCRIPTION
**Pull request by qinqon**
This clones kubevirt/kubevirtci repo and use the scripts there
adding the extra logic we need for nmstate

Documentation about how to integrate at projects is here:
https://docs.google.com/document/d/1miXACXwYbB6wZKGou3u-z6eaG8IyOSDLgqtLw03aNUQ/edit#heading=h.pj3hq3lrzbcd

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -101,23 +124,23 @@ cluster-sync-resources:
...
 	$(KUBECTL) delete --ignore-not-found -f $(local_handler_manifest)
 	$(KUBECTL) create -f $(local_handler_manifest)
 
-cluster-sync-manager: cluster-sync-resources $(local_manager_manifest)
-	IMAGE_REGISTRY=localhost:$(shell ./cluster/cli.sh ports registry | tr -d '\r') \
+cluster-sync-manager: cluster-sync-resources $(CLI) $(local_manager_manifest)
+	IMAGE_REGISTRY=localhost:$(shell $(CLI) ports registry | tr -d '\r') \
 		make push-manager
-	./cluster/cli.sh ssh node01 'sudo docker pull $(LOCAL_REGISTRY)/$(MANAGER_IMAGE_FULL_NAME)'
+	$(SSH) node01 'sudo docker pull $(LOCAL_REGISTRY)/$(MANAGER_IMAGE_FULL_NAME)'
```
Comment (SchSeba) [inline Makefile]: why we need this?

File: Makefile [hunk 2]
```diff
@@ -101,23 +124,23 @@ cluster-sync-resources:
 	fi
 
 
-cluster-sync-handler: $(local_handler_manifest)
-	IMAGE_REGISTRY=localhost:$(shell ./cluster/cli.sh ports registry | tr -d '\r') \
+cluster-sync-handler: $(local_handler_manifest) $(CLI) $(KUBECTL)
+	IMAGE_REGISTRY=localhost:$(shell $(CLI) ports registry | tr -d '\r') \
 		make push-handler
-	./cluster/cli.sh ssh node01 'sudo docker pull $(LOCAL_REGISTRY)/$(HANDLER_IMAGE_FULL_NAME)'
+	$(SSH) node01 'sudo docker pull $(LOCAL_REGISTRY)/$(HANDLER_IMAGE_FULL_NAME)'
```
Comment (SchSeba) [inline Makefile]: why we need this?

File: Makefile (hunk 1 above)
Comment (qinqon) [reply Makefile]: So we can change kubevirtci path just changing one line at the top of the Makefile.

File: Makefile (hunk 2 above)
Comment (qinqon) [reply Makefile]: So we can change kubevirtci path just changing one line at the top of the Makefile.

File: Makefile [hunk 3]
```diff
@@ -14,18 +14,28 @@ HANDLER_IMAGE ?= $(IMAGE_REGISTRY)/$(HANDLER_IMAGE_FULL_NAME)
...
+CLUSTER_DOWN ?= $(CLUSTER_DIR)/down.sh
+CLI ?= $(CLUSTER_DIR)/cli.sh
+SSH ?= $(CLI) ssh
 
 local_handler_manifest = build/_output/handler.local.yaml
 local_manager_manifest = build/_output/manager.local.yaml
 
 resources = deploy/service_account.yaml deploy/role.yaml deploy/role_binding.yaml
 
+foo:
```
[the remark this answers is not in the document]
Comment (qinqon) [reply Makefile]: Done

File: Makefile [hunk 4]
```diff
@@ -14,12 +14,19 @@ HANDLER_IMAGE ?= $(IMAGE_REGISTRY)/$(HANDLER_IMAGE_FULL_NAME)
...
 GINKGO_ARGS ?= -v -r --randomizeAllSpecs --randomizeSuites --race --trace $(GINKGO_EXTRA_ARGS)
 GINKGO?= go run ./vendor/github.com/onsi/ginkgo/ginkgo
-
-KUBECONFIG ?= ./cluster/.kubeconfig
-# TODO: use operator-sdk from vendor/
 OPERATOR_SDK ?= go run ./vendor/github.com/operator-framework/operator-sdk/cmd/operator-sdk
 LOCAL_REGISTRY ?= registry:5000
-KUBECTL ?= ./cluster/kubectl.sh
+
+export KUBEVIRT_PROVIDER ?= k8s-1.13.3
```
Comment (phoracek) [inline Makefile]: please check that our developer docs are up to date with this (especially the example of running OpenShift local cluster)

File: Makefile [hunk 5]
```diff
@@ -77,31 +84,38 @@ test/cluster/e2e:
...
+$(CLUSTER_DIR)/%: kubevirtci.version
+	rm -rf kubevirtci
+	git clone https://github.com/kubevirt/kubevirtci
+	cd kubevirtci && git checkout $$(cat  ../kubevirtci.version)
 
-cluster-up:
-	./cluster/up.sh
+cluster-up: $(CLUSTER_UP)
+	$(CLUSTER_UP)
+	$(SSH) node01 -- sudo yum install -y NetworkManager NetworkManager-ovs
```
Comment (phoracek) [inline Makefile]: here you use `-- command`, while in other targets ` 'command'`. Please stick to one.

File: Makefile [hunk 6]
```diff
@@ -77,31 +84,38 @@ test/cluster/e2e:
...
+	$(SSH) node01 -- sudo systemctl daemon-reload
+	$(SSH) node01 -- sudo systemctl restart NetworkManager
 
-cluster-down:
-	./cluster/down.sh
+cluster-down: $(CLUSTER_DOWN)
+	$(CLUSTER_DOWN)
 
-cluster-clean:
+cluster-clean: $(KUBECTL)
```
Comment (phoracek) [inline Makefile]: the kubectl logic is neat!

File: Makefile (hunk 4 above)
Comment (qinqon) [reply Makefile]: Fixed, going back to 1.11.0 I was all wrong with that.

File: Makefile (hunk 5 above)
Comment (qinqon) [reply Makefile]: It has to be -- command is how it used elsewhere and just 'command' fails.

File: Makefile (hunk 6 above)
Comment (qinqon) [reply Makefile]: We have to remember to add target dependency though.

File: Makefile [hunk 7]
```diff
@@ -100,9 +100,7 @@ $(CLUSTER_DIR)/%: kubevirtci.version
 
 cluster-up: $(CLUSTER_UP)
 	$(CLUSTER_UP)
-	$(SSH) node01 -- sudo yum install -y NetworkManager NetworkManager-ovs
-	$(SSH) node01 -- sudo systemctl daemon-reload
-	$(SSH) node01 -- sudo systemctl restart NetworkManager
+	hacks/install-nm.sh
```
Comment (phoracek) [inline Makefile]: we already have hack/

File: hacks/install-nm.sh [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+node01_id=$(docker ps |grep node01 |awk '{print $1}')
+
+echo 'Install NetworkManager on the node'
+docker exec $node01_id ssh.sh sudo yum install -y NetworkManager
+docker exec $node01_id ssh.sh sudo systemctl daemon-reload
+docker exec $node01_id ssh.sh sudo systemctl restart NetworkManager
+
+echo 'Check NetworkManager is working fine'
+docker exec $node01_id ssh.sh nmcli device show > /dev/null
+
```
Comment (phoracek) [inline hacks/install-nm.sh]: drop nasty trailing line please

File: hacks/install-nm.sh [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash -e
+
+# TODO: Iterate all the nodes
```
Comment (phoracek) [inline hacks/install-nm.sh]: this will happen only with multi nodes PR, right?

File: Makefile (hunk 7 above)
Comment (qinqon) [reply Makefile]: Done

File: hacks/install-nm.sh (hunk 8 above)
Comment (qinqon) [reply hacks/install-nm.sh]: Done

File: hacks/install-nm.sh (hunk 9 above)
Comment (qinqon) [reply hacks/install-nm.sh]: Yep, let's fix this there, the idea is to test at jenkins with multiple nodes (We have to make test transparent to the number of nodes).

File: test/e2e/nodes_test.go [hunk 10]
```diff
@@ -39,7 +39,6 @@ var _ = Describe("Nodes", func() {
 				obtainedInterfaces := interfacesName(interfaces)
 				Expect(obtainedInterfaces).To(SatisfyAll(
 					ContainElement("eth0"),
-					ContainElement("eth1"),
```
Comment (phoracek) [inline test/e2e/nodes_test.go]: there is a major problem with this. everyone has eth0, even a simple pod. hopefully will be fixed once we are able to create interfaces (or request on local cluster)

File: Makefile (hunk 2 above)
Comment (qinqon) [reply Makefile]: Let's drop it.

**Pull request merged**